pulp_container/app/viewsets.py

```diff
@@ -611,7 +611,7 @@ class ContainerDistributionViewSet(BaseDistributionViewSet):
                 "effect": "allow",
                 "condition": [
                     "has_model_or_obj_perms:container.view_containerdistribution",
-                    "has_view_namespace_perms",
+                    # "has_namespace_obj_perms:container.view_containernamespace",
```

Comment:
I am not sure it is correct to require `has_namespace_obj_perms:container.view_containernamespace` if I already can view the distribution whether the one I have just created or I have a model perm to view all dists.

Reply:
You can add a second statement for "retrieve" that allows to view on the "manage_namespace_distributions" permission. Or we combine that kind of logic in a new `has_model_or_obj_or_namespace_perms`.
The latter cannot easily be adjusted by admins, however with the former it's much easier to screw up.